docs/userguides/networks.md

```diff
@@ -98,6 +98,98 @@ In the remainder of this guide, any example below using Ethereum, you can replac
+
+To add custom networks to your `ape-config.yaml` file, follow this pattern:
+
+```yaml
+networks:
+  custom:
+     - name: apenet             # Required
+       chain_id: 9867733322210  # Required
+       ecosysem: ethereum       # Defaults to your default ecosystem
+       default_provider: geth   # Default is the generic node provider
+```
+
+The following paragraphs explain the different parameters of the custom network config.
+
+**name**: The `name` of the network is the same identifier you use in the network triplet for the "network" (second) section.
+Read more on the network option [here](#selecting-a-network).
+
+**chain_id**: The chain ID is required for config-based custom networks.
+It ensures you are on the correct network when making transactions and is very important!
+
+**ecosystem**: You can optionally change the ecosystem class Ape uses.
+The ecosystem system class is largely responsible for decoding and encoding data to-and-fro the blockchain.
+More information about the EcosystemAPI can be found [here](../methoddocs/api.html#ape.api.networks.EcosystemAPI).
+The default ecosystem is `ethereum`, which is the base class to most other L2 ecosystems and is defined in the core plugin `ape-ethereum`.
+If your custom network's ecosystem matches closer to another L2 instead of Ethereum, use that ecosystem name as your `ecosystem` in your custom network config.
+For example, take note that `"ethereum"` assumes EIP-1559 exists (unless configured otherwise).
+If your custom network is closer to Fantom, Polygon, Avalanche, or any other L2, you may want to consider using one of those plugins as the `ecosystem` to your custom network.
+Alternatively, you can configure your custom network the same way you configure any other network in the config (see [this section](#block-time-transaction-type-and-more-config)).
+
+**default_provider**: The default provider is the provider class used for making the connection to your custom network, unless you specify a different provider (hence the `default_`).
+Generally, you won't change this and can use the default EVM node provider.
+Many provider plugins won't function here, such as `ape-infura` or `ape-alchemy`.
+If you are using one of their networks, it is best to edit and use the plugins directly.
+If you are using a developer-node remotely, such as a custom Anvil node, you can specify the default provider to be `foundry` instead.
+However, take care in making sure you set up Foundry to correctly connect to your node.
+Likewise, when using the default Ethereum node provider, you will need to tell it the RPC URL.
+
+#### RPC URL
+
+To configure the RPC URL for a custom network, use the configuration of the provider.
+For example, if the RPC URL is `https://apenet.example.com/rpc`, configure it by doing:
+
+```yaml
+default_ecosystem: ethereum  # Is default, but including for clarity.
+
+networks:
+  custom:
+     - name: apenet
```

Comment:
question: What would happen if two networks are defined with the same name?

suggestion: Maybe this would be better as a dict.

Reply:
They can have the same name if they are in different ecosystems.
For example, almost every ecosystem has a network named "mainnet".

I will add a test for this case and see what happens and make sure it is handled ok

Reply:
https://github.com/ApeWorX/ape/pull/1852/commits/66a3bc6c87cd6e4555fc28fd531e4fe8fc7e37e2

and we can't use dict in case using different ecosystems